docs/adrs/002-change-release-strategy.md

```diff
@@ -0,0 +1,78 @@
+---
+status: proposed
+date: 17-05-2024
+deciders: Kamil Kołodziej, Radosław Szuma, Jakub Pisarek
+informed: Fishjam Team, Cloud Team
+---
+# Change release strategy to eliminate downtimes
+
+## Context and Problem Statement
+
+In the current situation when we do a release we are shuting down the server and every room with that.
+This is an awful user experience. Another issue is the fact that any ongoing recordings are lost.
+
+## Decision Drivers
+
+* N o downtime while release happens
+* 0 lost recordings during release
+
+## Considered Options
+
+* Rolling update with eviction
+* Active room migration
+
+## Decision Outcome
+
+Chosen option: "Rolling update with eviction", because
+while second option sounds great, we acknowledge that solution is not trivial and we want to fix downtimes ASAP. First solution is a good starting point to remove existing problems and allow us to move forward.
+
+### Consequences
+
+Every new release will need to trigger fishjam process which will handle the shutdown. What that effectively means is that we are leaving the responsibility of triggering release to external orchestration tool (what is usually the case) and to monitor if the application exited correctly.
```

Comment:
**issue**: So, release != deployment or upgrade. I suggest generally replacing this word everywhere in this ADR with something else. IMHO it makes it harder to understand, as this has not much in common with the Fishjam release cycle.

```suggestion
Every new deployment will need to trigger fishjam process which will handle the shutdown. What that effectively means is that we are leaving the responsibility of triggering release to external orchestration tool (what is usually the case) and to monitor if the application exited correctly.
```

Reply:
Seconding this, in favour of 'deployment' as it can refer to both an 'upgrade' and a 'downgrade'/'downpatch'